File: original_recipe/bbcursive/src/main/java/bbcursive/lib/pos.java
package bbcursive.lib;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.nio.ByteBuffer;
import java.util.function.UnaryOperator;

/**
 * Created by jim on 1/17/16.
 */
public class pos implements UnaryOperator<ByteBuffer> {
    private final int position;

    public pos(int position) {
        this.position = position;
    }

    /**
     * reposition
     *
     * @param position
     * @return
     */
    @NotNull
    public static UnaryOperator<ByteBuffer> pos(int position) {
        return new pos(position){
            @NotNull
            @Override
            public String toString() {
                return "pos("+position+")";
            }
        };
    }

    @NotNull
    @Override
    public ByteBuffer apply(@Nullable ByteBuffer t) {
        return null == t ? t : t.position(position);
    }
}

File: original_recipe/bbcursive/src/main/java/bbcursive/lib/pos.kt
package bbcursive.lib

import bbcursive.func.UnaryOperator
import bbcursive.std
import java.nio.ByteBuffer

/**
 * Created by jim on 1/17/16.
 */
open class pos(private val position: Int) : UnaryOperator<ByteBuffer> {


    override fun invoke(t: ByteBuffer): ByteBuffer {
        return if (std.NULL_BUFF == t) t else t.position(position)
    }

    override fun toString(): String {
        return "pos($position)"
    }
}